File: src/Application Specific/Classes/Entity Collections/EngineerPostalRegion/EngineerPostalRegionValidator.vb
Namespace Entity

    Namespace EngineerPostalRegions

        Public Class EngineerPostalRegionValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oEngineerPostalRegion As EngineerPostalRegion)

                'make sure that contact object is valid
                If oEngineerPostalRegion.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oEngineerPostalRegion.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If



                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/EngineerPostalRegion/EngineerPostalRegionValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace EngineerPostalRegions
    {
        // make sure that contact object is valid
        public class EngineerPostalRegionValidator : BaseValidator
        {
            public void Validate(EngineerPostalRegion oEngineerPostalRegion)
            {

                // make sure that contact object is valid
                if (oEngineerPostalRegion.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oEngineerPostalRegion.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}